### PR TITLE
Issue#122: auto create app from part of label

### PR DESCRIPTION
**Pull request by amylin1**
PR for issue#122 https://github.com/kappnav/issues/issues/122

### CODE REVIEW

File: auto_create.go [hunk 1]
```diff
@@ -52,6 +52,12 @@ const (
 	defaultAutoCreateAppVersion = "1.0.0"
 	defaultAutoCreateAppLabel   = "app"
 
+	// AppPartOf ...
```
Comment (cvignola) [inline auto_create.go:55]: Nit:  seems unnecessary to have two constants with same value.

File: auto_create.go [hunk 2]
```diff
@@ -212,23 +246,25 @@ func (resController *ClusterWatcher) parseAutoCreateResourceInfo(unstructuredObj
...
 		} else {
 			arrayOfString := stringToArrayOfAlphaNumeric(autoCreateKindsStr)
 			if len(arrayOfString) == 0 {
 				if logger.IsEnabled(LogTypeError) {
 					logger.Log(CallerName(), LogTypeError, fmt.Sprintf("Metadata %s for resource %s/%s does not contain comma separated kinds. Its current value is %s. Switching to default.", AppAutoCreateKinds, resourceInfo.resourceInfo.namespace, resourceInfo.resourceInfo.name, autoCreateKindsStr))
-				}
-				resourceInfo.autoCreateKinds = defaultKinds
+				}				
+				resourceInfo.autoCreateKinds = []groupKind{
+					groupKind{group: "apps", kind: resourceInfo.kind}}
```
Comment (cvignola) [inline auto_create.go]: The group value "apps" will not work for all resource kinds.  It only works for "core" kinds for backward compatibility.  The group value should be obtained from resource's apiVersion field with this algorithm:

if apiVersion contains group name, use it
else use group name "core"

e.g. 
```
apiVersion: v1 ----> group name is "" 
apiVersion: extensions/v1beta1 ---> group name is "extensions" 
apiVersion: openliberty.io/v1beta1 ---> group name is "openliberty.io" 
```

**Review by cvignola — CHANGES_REQUESTED** [review]

File: auto_create.go (hunk 1 above)
Comment (amylin1) [reply auto_create.go:55]: done

File: auto_create.go (hunk 2 above)
Comment (amylin1) [reply auto_create.go]: it is for debugging purpose and I forgot to clean up it.  It uses default component kinds in the existing code when no auto-create-kinds label specified.

**Review by cvignola — APPROVED** [review]

**Pull request merged**